enhancedTable/panel-rows-manager.ts

```diff
@@ -21,16 +21,34 @@ export class PanelRowsManager {
         this.initTableRowVisibility();
 
         // Subscribers
-        // Filter all rows when visibility is off
+        // Filter all rows when visibility is off and none when it's on
+        // Requires a separate check since it's not handled as a filter change
         this.layerVisibilityObserver = this.legendBlock.visibilityChanged.subscribe(visibility => {
             this.validOids = visibility ? undefined : [];
             this.externalFilter = !visibility;
             this.updateGridFilters();
         });
 
-        // Fetch valid rows and filter table on symbol toggle
-        this.symbolVisibilityObserver = this.legendBlock.symbolVisibilityChanged.subscribe(() => {
-            this.fetchValidOids(this.extent);
+        // Update table on map filter change
+        this.mapFilterChangedObserver = this.mapApi.filterChanged.subscribe((params) =>  {
+            const filterTypes = this.legendBlock.proxyWrapper.filterState.coreFilterTypes;
+            switch(params.filterType) {
+                case filterTypes.EXTENT: // Filter table by extent if enabled
+                    this.extent = params.extent;
+                    if (this.panelManager.filterByExtent) {
+                        this.fetchValidOids(this.extent);
+                    }
+                    break;
+
+                case filterTypes.SYMBOL: // Filter table by symbols
```

Comment:
I don't think we should be explicitly checking for `SYMBOL` here.  It should react to any not-`EXTENT`, not-`GRID` filter that matches the layer keys.